testplan/runners/pools/base.py

```diff
@@ -213,11 +227,22 @@ def _loop(self, transport: QueueClient) -> None:
             elif received.cmd == Message.TaskSending:
                 results = []
                 for item in received.data:
-                    results.append(self.execute(item))
-                transport.send_and_receive(
+                    if self._discard_running.is_set():
+                        break
+
+                    task_result = self.execute(item)
+                    if not self._discard_running.is_set():
```

Comment:
I kind of don't like this exception. Can we have execute always return the result?

Reply:
maybe we can make a better usage of TaskResult.status, currently it's just a boolean flag